### PR TITLE
feat(query): added "Function App Authentication Disabled" for Terraform

### PR DESCRIPTION
**Pull request by rafaela-soares**
**Proposed Changes**
- Added "Function App Authentication Disabled" for Terraform
- Renamed folder from assets\queries\terraform\azure\unrestricted_sql_server_acess to assets\queries\terraform\azure\unrestricted_sql_server_access

I submit this contribution under the Apache-2.0 license.

### CODE REVIEW

**Review by JoaoDuarteGomes — CHANGES_REQUESTED** [review]
Category LGTM.
Severity should be HIGH.

File: assets/queries/terraform/azure/function_app_authentication_disabled/metadata.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "id": "e65a0733-94a0-4826-82f4-df529f4c593f",
+  "queryName": "Function App Authentication Disabled",
+  "severity": "INFO",
```
Comment (rogeriopeixotocx) [inline assets/queries/terraform/azure/function_app_authentication_disabled/metadata.json:4]:
```suggestion
  "severity": "HIGH",
```

**Review by rogeriopeixotocx — CHANGES_REQUESTED** [review]

**Review by JoaoDuarteGomes — APPROVED** [review]
LGTM.

**Review by joaoReigota1 — APPROVED** [review]
LGTM

**Pull request merged**